lib/pos_blockchain.go

```diff
@@ -537,6 +547,29 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 	return errors.Wrap(err, "processBlockPoS: Problem adding block to block index: ")
 }
 
+// checkAndStoreArchivalBlock is a helper function that takes in a block and checks if it's an archival block.
+// If it is, it stores the block in the db and returns true. If it's not, it returns false, or false and an error.
+func (bc *Blockchain) checkAndStoreArchivalBlock(block *MsgDeSoBlock) (_success bool, _err error) {
+	// First, get the block hash and lookup the block index.
+	blockHash, err := block.Hash()
+	if err != nil {
+		return false, errors.Wrap(err, "checkAndStoreArchivalBlock: Problem hashing block")
+	}
+	blockNode, exists := bc.blockIndexByHash[*blockHash]
+	// If the blockNode doesn't exist, or the block is not committed, or it's already stored, then we're not dealing
+	// with an archival block. Archival blocks must have an existing blockNode, be committed, and not be stored.
+	if !exists || !blockNode.IsCommitted() || blockNode.IsStored() {
+		return false, nil
+	}
```

Comment:
```suggestion
	if !exists || !blockNode.IsCommitted() {
		return false, errors.New("checkAndStoreArchivalBlock: block is not committed")
	}
	
	// If the block is already stored, then there's nothing left to do.
	if blockNode.isStored() {
		return true, nil
	}
```

Reply:
I feel like returning an error here could be problematic. If a blocknode exists, and a block is not committed, I think we should continue with the remainder of `processBlockPoS`. Instead, if we error here, we would break out of that function, which seems undesirable.